pyemu/utils/pst_from.py

```diff
@@ -150,25 +152,10 @@ def parfile_relations(self):
                 else pd.Series(
                     {k: v for k, v in [['ubound', x.upper_bound]]}), axis=1)
             if ubound.nunique(0, False).gt(1).any():
-                if ubound.nunique(0, False).gt(2).any():
-                    # more than one upper bound set
-                    self.logger.lraise(
-                        "different upper bounds requested for same par for {0}"
-                        "".format(name))
-                else:
-                    # one set - the rest are None - need to replace None
-                    # with set values
-                    # df with set values
-                    fil = ubound.apply(lambda x:
-                                       pd.Series([None]) if x.isna().all()
-                                       else x[x.notna()].values).T
-                    self.logger.warn("Upper bound for par passed for some but "
-                                     "not all instances, will set NA to "
-                                     "passed values\n{}".format(fil))
-                    # replace Nones in list in Series with passed values
-                    pr.loc[g.index, 'upper_bound'] = g.use_cols.apply(
-                        lambda x: [fil[0].loc['ubound{0}'.format(c)] for c in x]
-                        if x is not None else fil[0].loc['ubound'])
+                ub_min = ubound.min().fillna(self.ult_ubound_fill).to_dict()
```

Comment:
Wonder if min() and fillna() need to be the other way around? Can't remember if pandas returns na as the minimum if the series contains na, and we want to make sure the 'None's are set to a value if one is ever passed. NB. Are we happy skipping through with the min passed value if the user specifies more than one bound for a given par?

Reply:
I thought about this too. I  settled on reduction then fill because that means we sort out which user-supplied value to use, then fill any of the remaining all-nan entries with the fill value.  Im not sure which of min/max to use for upper/lower so I just picked min-upper and max-lower so that we dont choose extreme values over more meaningful values.

Reply:
Yeah, that works, looks like pandas ignores NaNs for min/max operations (differs from numpy here). min-upper, max-lower makes sense to me.